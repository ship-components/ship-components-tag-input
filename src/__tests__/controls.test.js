import React from 'react';
import Immutable from 'immutable';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SelectControls from '../Controls';

function createSampleProps() {
  return {
    darkTheme: true,
    filterable: true,
    label: 'sample',
    orderOptionsBy: 'title',
    selection: new Immutable.List(),
    onFocus: jest.fn(),
    onChange: jest.fn(),
    onKeyDown: jest.fn(),
    onClear: jest.fn(),
    onToggle: jest.fn()
  };
}

// Creates a shallow copy of SelectControls
function initializeShallowTagInputComponent() {
  const props = createSampleProps();

  return shallow(
    <SelectControls {...props} />
  );
}

// Creates a mounted copy of SelectControls
function initializeMountTagInputComponent() {
  const props = createSampleProps();

  return mount(
    <SelectControls {...props}/>
  );
}

describe('Component: Controls', () => {
  describe('SelectControls Rendering', () => {
    it('should render correctly', () => {
      const tree = initializeShallowTagInputComponent();
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });

  describe('focusInput Function', () => {
    it('should fires a focus event when focusInput function calls', () => {
      const wrapper = initializeMountTagInputComponent();
      const textInputWrapper = wrapper.find('input [type="text"]');
      const textInput = textInputWrapper.get(0);
      const focusInputSpy = jest.spyOn(textInput, 'focus');

      // Fires the focusInput function
      wrapper.node.focusInput();

      expect(focusInputSpy).toHaveBeenCalled();
    });

    it('should NOT fires a focus event when filterable is false', () => {
      const wrapper = initializeMountTagInputComponent();
      const textInputWrapper = wrapper.find('input [type="text"]');
      const textInput = textInputWrapper.get(0);
      const focusInputSpy = jest.spyOn(textInput, 'focus');

      // Update filterable to be false
      wrapper.setProps({
        filterable: false
      });

      // Fires the focusInput function
      wrapper.node.focusInput();

      expect(focusInputSpy).not.toHaveBeenCalled();
    });
  });

  describe('blurInput Function', () => {
    it('should fires a blur event when blurInput function calls', () => {
      const wrapper = initializeMountTagInputComponent();
      const textInputWrapper = wrapper.find('input [type="text"]');
      const textInput = textInputWrapper.get(0);
      const blurInputSpy = jest.spyOn(textInput, 'blur');

      // Fires the blurInput function
      wrapper.node.blurInput();

      expect(blurInputSpy).toHaveBeenCalled();
    });

    it('should NOT fires a blur event when filterable is false', () => {
      const wrapper = initializeMountTagInputComponent();
      const textInputWrapper = wrapper.find('input [type="text"]');
      const textInput = textInputWrapper.get(0);
      const blurInputSpy = jest.spyOn(textInput, 'blur');

      // Update filterable to be false
      wrapper.setProps({
        filterable: false
      });

      // Fires the blurInput function
      wrapper.node.blurInput();

      expect(blurInputSpy).not.toHaveBeenCalled();
    });
  });
});
