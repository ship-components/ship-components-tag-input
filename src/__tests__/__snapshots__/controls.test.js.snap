// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Controls SelectControls Rendering should render correctly 1`] = `
<div>
  <div>
    <button
      className=""
      onClick={[MockFunction]}
      tabIndex="-1"
    >
      library_add
    </button>
    <Loader
      absolute={false}
      className=""
      spinnerClassName=""
      visible={false}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <span
      className=""
    />
    <input
      className="hidden"
      onChange={[MockFunction]}
      onKeyDown={[MockFunction]}
      type="text"
      value=""
    />
  </div>
</div>
`;
